src/web_info.rs

```diff
@@ -0,0 +1,43 @@
+use crate::error::AybError;
+use crate::templating::TemplateString;
+use serde::Deserialize;
+use url::Url;
+
+#[derive(Clone, Deserialize)]
+pub struct WebInfo {
+    base_url: Url,
+    endpoints: WebEndpoints,
+}
+
+#[derive(Clone, Deserialize)]
+pub struct WebEndpoints {
+    profile: TemplateString,
+    confirmation: TemplateString,
+}
+
+impl WebInfo {
+    pub async fn from_url(url: &Url) -> Result<Self, AybError> {
+        Ok(reqwest::get(url.to_string()).await?.json().await?)
```

Comment:
Given the templates evaluate the same way each time, can we push the logic of running `profile()`/`confirmation()` into the constructor, make the struct fields `pub`, and make the methods private? That way we don't have to execute a template and compute a URL each time we need it.